orbstation/quirks/plurality.dm

```diff
@@ -0,0 +1,316 @@
+
+// Plural system chip item, used to store information on plural systems.
+
+/obj/item/plural_system_chip
+	name = "plural system chip"
+	desc = "A chip used to store data on plural system members. Can be attached to an \
+	identification card in order to allow it to switch between the names of the members, \
+	and removed from the card with a screwdriver. A Head of Personnel's ID can be swiped on \
+	the chip to lock and unlock it, allowing users to edit, add, or remove members to or from \
+	the system."
+	icon = 'orbstation/icons/obj/misc.dmi'
+	icon_state = "plural_chip_1"
+	w_class = WEIGHT_CLASS_TINY
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, FIRE = 100, ACID = 100)
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	req_access = list(ACCESS_HOP)
+
+	/// Whether the chip is locked or unlocked. When unlocked, it can be used in-hand to edit/add/remove system member names.
+	var/locked = TRUE
+	/// The name of the plural system, which is displayed on the ID card in {braces} at all times.
+	var/system_name
+	/// The name of the system's current fronter, which will be displayed on the ID card in place of the registered name.
+	var/fronter_name
+	/// List of system member names. When the chip is attached to an ID, using the ID in-hand will allow the user to switch between these names.
+	var/list/system_member_names = list()
+	/// When TRUE, prevents the system name from being modified. Used for the roundstart chips given by the plurality quirk. Set to FALSE when a HoP ID is swiped on the chip.
+	var/system_name_locked = FALSE
+
+/obj/item/plural_system_chip/update_icon_state()
+	icon_state = "plural_chip"
+	icon_state += "_[locked]"
+	return ..()
+
+/obj/item/plural_system_chip/attack_self(mob/user)
+	if(user.incapacitated())
+		return
+	if(!user.is_holding(src))
+		return
+	if(locked)
+		to_chat(user, span_warning("[src] is locked and cannot be modified right now. It can be unlocked by an ID card with Head of Personnel access."))
+		balloon_alert(user, "locked")
+		return
+	if(!system_name)
+		var/new_system_name
+		if(system_name_locked)
+			new_system_name = user.mind.name ? user.mind.name : user.name
+			update_system_name(new_system_name)
+			to_chat(user, span_notice("The chip's associated system name was automatically set to [new_system_name]."))
+			// no return here because we're jumping right into the Choose Action menu
+		else
+			new_system_name = tgui_input_text(user, "Input the new system name:", "Set System Name", max_length = 42)
+			if(!new_system_name)
+				return
+			update_system_name(new_system_name)
+			to_chat(user, span_notice("Successfully set the chip's associated system name to [new_system_name]."))
+			return
+	var/popup_input = tgui_input_list(user, "Choose Action", "Plural System Interface", list("Add Member", "Remove Member", "Edit Member", "Clear Members", "Edit System Name"))
+	if(popup_input == "Add Member")
```

Comment:
All of these ` if(popup_input == "")` look like they should be a switch statement, for both clarity and also as a minor optimisation (switches jump between options slightly faster).